### DISCUSSION TITLE
How can I always build from source?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#1671] [Everyday usage] [answered]
I am using macOS 12 and brew forces me to build from source everything and I am totally ok with that, but I don't want brew to always ask me to build from source e.g. I don't want this:

```
❯ brew install -v htop --head
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
Updated 4 formulae.
==> Updated Casks
Updated 1 cask.

Warning: You are using macOS 12.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
pre-release version.

Error: autoconf: no bottle available!
You can try to install from source with:
  brew install --build-from-source autoconf
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```
but I want brew to automatically build deps and packages from source(e.g. autoconf and htop for this example) in these conditions so I won't need to write `brew install --build-from-source *dependency*` every time.

Is there a config file or a tweak to make this possible?

**Comment by nandahkrishna** [comment, marked as answer]
You don't need to build everything from source, please take a look at this comment: https://github.com/Homebrew/brew/issues/11523#issuecomment-858839837. Bottles from previous versions of macOS will be used instead.

**Reply by ghost** [reply]
but what if I want to? Just curious

**Reply by SMillerDev** [reply]
Then you'll have to hack support into brew for that since it's an unsupported configuration.
